src/zenml/zen_stores/schemas/user_schemas.py

```diff
@@ -200,3 +257,33 @@ def to_model(
                 updated=self.updated,
                 roles=[ra.role.to_model() for ra in self.assigned_roles],
             )
+
+    def to_service_account_model(
```

Comment:
There are guards in the SQL zen store to prevent against such a thing (and integration tests for it). Failing here would be redundant.

Reply:
The reverse is not true: `to_model()` is still called in lots of places to hydrate the owner of various resources in a `UserResponseModel` format. I have to allow that, otherwise massive code changes would be required in both core and dashboard to allow an `owner` to be either `UserResponseModel` or `ServiceAccountResponseModel`.